src/EnergyPlus/MixedAir.cc

```diff
@@ -5436,6 +5436,76 @@ namespace MixedAir {
 	}
 
 	int
+	GetOASysNumHXs( int const OASysNumber ) 
+	{
+
+		// FUNCTION INFORMATION:
+		//       AUTHOR         Fred Buhl, Rongpeng Zhang
+		//       DATE WRITTEN   Oct. 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS FUNCTION:
+		// After making sure get input is done, the number of heat recovery exchangers in the
+		// OA System is returned.
+
+		// METHODOLOGY EMPLOYED:
+		// na
+
+		// REFERENCES:
+		// na
+
+		// USE STATEMENTS:
+
+		// Return value
+		int NumHX; // number of heat exchangers in this OA System
+
+		// Locals
+		// FUNCTION ARGUMENT DEFINITIONS:
+
+		// FUNCTION PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
+
+		// FUNCTION LOCAL VARIABLE DECLARATIONS:
+		std::string CompType;
+		std::string CompName;
+		bool Sim;
+		bool FirstHVACIteration;
+		bool OAHeatingCoil;
+		bool OACoolingCoil;
+		int CompNum;
+		int AirLoopNum;
+		bool OAHX;
+
+		if( GetOASysInputFlag ) {
+			GetOutsideAirSysInputs();
+			GetOASysInputFlag = false;
+		}
+
+		Sim = false;
+		FirstHVACIteration = false;
+		AirLoopNum = 0;
+		NumHX = 0;
+		for( CompNum = 1; CompNum <= OutsideAirSys( OASysNumber ).NumComponents; ++CompNum ) {
+			CompType = OutsideAirSys( OASysNumber ).ComponentType( CompNum );
+			CompName = OutsideAirSys( OASysNumber ).ComponentName( CompNum );
+			SimOAComponent( CompType, CompName, OutsideAirSys( OASysNumber ).ComponentType_Num( CompNum ), FirstHVACIteration, OutsideAirSys( OASysNumber ).ComponentIndex( CompNum ), AirLoopNum, Sim, OASysNumber, OAHeatingCoil, OACoolingCoil, OAHX );
```

Comment:
@rongpengzhang Why do you need to call SimOAComponent to set OAHX to then increment NumHX counter? This seems, on the surface, to be an inefficient way to find the number of OA HX. Looking at the `SimOAComponent` code, it seems like the `Sim` boolean makes it so the function basically doesn't do anything, which I think is an odd input parameter for a function called `SimOAComponent`.  This may be the best way to find the number of OA HXs, but it is not intuitive to me. For a seemingly, purpose build function to `GetOASysNumHXs` I think I would take that functionality out of `SimOAComponent` and put it in here.

```cpp
for( CompNum = 1; CompNum <= OutsideAirSys( OASysNumber ).NumComponents; ++CompNum ) {
  int componentTypeNum = OutsideAirSys( OASysNumber ).ComponentType_Num( CompNum );
  if ( HeatXchngr == componentTypeNum || Desiccant == componentTypeNum ) {
    +NumHX;
  }
}
```

But that is my opinion, others can weigh in if they wish.

On a side note, @DeadParrot I was unaware that `Optional_bool` is a reference, or at least appears to behave like one from this code. Is this true?